rethinkdb/tests/conftest.py

```diff
@@ -0,0 +1,46 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Iterator
+
+import pytest
+
+from datadog_checks.dev import docker_run
+from datadog_checks.rethinkdb.types import Instance
+
+from .cluster import setup_cluster
+from .common import (
+    AGENT_PASSWORD,
+    AGENT_USER,
+    BOOTSTRAP_SERVER,
+    COMPOSE_ENV_VARS,
+    COMPOSE_FILE,
+    HOST,
+    SERVER_PORTS,
+    SERVERS,
+)
+
+
+@pytest.fixture(scope='session')
+def instance():
+    # type: () -> Instance
+    return {
+        'host': HOST,
+        'port': SERVER_PORTS['server0'],
+        'username': AGENT_USER,
+        'password': AGENT_PASSWORD,
+    }
+
+
+@pytest.fixture(scope='session')
+def dd_environment(instance):
+    # type: (Instance) -> Iterator
+    conditions = [setup_cluster]
+
+    log_patterns = [r'Server ready, "{}".*'.format(BOOTSTRAP_SERVER)]
+    log_patterns += [r'Connected to server "{}".*'.format(server) for server in SERVERS - {BOOTSTRAP_SERVER}]
+    log_patterns += [r'Connected to proxy.*']
+
+    with docker_run(COMPOSE_FILE, conditions=conditions, env_vars=COMPOSE_ENV_VARS, log_patterns=log_patterns):
```

Comment:
About `log_patterns`, by default it will stop waiting when one of the `log_patterns` is matched. Is that the intent or you want to match all the patterns?

Reply:
Oh, not intended no — I thought it would match all patterns and then carry on. In practice RethinkDB is super fast to start up though, so I never went across any issues. Is there a way to have it wait for all `log_patterns` before continuing?

Reply:
You can do something like this:

```

@pytest.fixture(scope="session")
def dd_environment():
    compose_file = os.path.join(HERE, 'compose', 'docker-compose.yaml')
    with docker_run(
        compose_file,
        conditions=[
            # Kafka Broker
            CheckDockerLogs('broker', 'Monitored service is now ready'),
            # Kafka Schema Registry
            CheckDockerLogs('schema-registry', 'Server started, listening for requests...', attempts=90),
            # Kafka Connect
            CheckDockerLogs('connect', 'Kafka Connect started', attempts=120),
        ],
    ):
        yield CHECK_CONFIG, {'use_jmx': True}
```

This is just a shortcut:

```
    with docker_run(COMPOSE_FILE, conditions=conditions, env_vars=COMPOSE_ENV_VARS, log_patterns=log_patterns):

```

Reply:
`matches='all'` https://github.com/DataDog/integrations-core/pull/5984/files

Reply:
Thanks! Addressed in 467e126